api/tests/Security/JWT/ValidatingSplitCookieExtractorTest.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace App\Tests\Security\JWT;
+
+use App\Security\JWT\ValidatingSplitCookieExtractor;
+use Lexik\Bundle\JWTAuthenticationBundle\TokenExtractor\TokenExtractorInterface;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * @internal
+ */
+class ValidatingSplitCookieExtractorTest extends TestCase {
+    public const HEADER = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXQSJ9';
+    public const DATA = 'eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ';
+    public const SIGNATURE = 'peuvjFPw8tuUhdkmCRV-a1qVB7yYQE2a0dFuBcrv0DM';
+
+    public const JWT_HP = self::HEADER.'.'.self::DATA;
+    public const JWT_S = self::SIGNATURE;
+
+    private MockObject|TokenExtractorInterface $decorated;
+    private ValidatingSplitCookieExtractor $validatingSplitCookieExtractor;
+    private $request;
+
+    protected function setUp(): void {
+        $this->decorated = $this->createMock(TokenExtractorInterface::class);
+        $this->validatingSplitCookieExtractor = new ValidatingSplitCookieExtractor($this->decorated);
+        $this->request = new Request();
+    }
+
+    public function testReturnValidConcatenatedJWTToken() {
+        $jwt = self::JWT_HP.'.'.self::JWT_S;
+        $this->decorated->expects(self::once())->method('extract')->willReturn($jwt);
+
+        $result = $this->validatingSplitCookieExtractor->extract($this->request);
+
+        self::assertThat($result, self::equalTo($jwt));
+    }
+
+    public function testReturnFalseWhenDecoratedReturnsFalse() {
+        $this->decorated->expects(self::once())->method('extract')->willReturn(false);
+
+        $result = $this->validatingSplitCookieExtractor->extract($this->request);
+
+        self::assertThat($result, self::isFalse());
+    }
+
+    public function testReturnFalseWhenDecoratedOnlyReturnsJWTHP() {
+        $this->decorated->expects(self::once())->method('extract')->willReturn(self::JWT_HP);
+
+        $result = $this->validatingSplitCookieExtractor->extract($this->request);
+
+        self::assertThat($result, self::isFalse());
+    }
+
+    public function testReturnFalseWhenDecoratedOnlyReturnsJWTS() {
+        $this->decorated->expects(self::once())->method('extract')->willReturn(self::JWT_S);
```

Comment:
Given our current configuration, `decorated` would never return a string formatted like `self::JWT_S`, but rather `'.' . self::JWT_S`. And that is exactly the case which the current implementation of ValidatingSplitCookieExtractor doesn't cover.

Reply:
done